src/Compilers/CSharp/Portable/Symbols/MemberSymbolExtensions.cs

```diff
@@ -154,6 +155,99 @@ internal static TMember ConstructIncludingExtension<TMember>(this TMember member
             throw ExceptionUtilities.UnexpectedValue(member);
         }
 
+        // For lookup APIs in the semantic model, we can return symbols that aren't fully inferred.
+        // But for function type inference, if the symbol isn't fully inferred with the information we have (the receiver and any explicit type arguments)
+        // then we won't return it.
+        internal static Symbol? GetReducedAndFilteredSymbol(this Symbol member, ImmutableArray<TypeWithAnnotations> typeArguments, TypeSymbol receiverType, CSharpCompilation compilation, bool checkFullyInferred)
```

Comment:
:memo: This logic is extracted from `AddReducedAndFilteredSymbol`, but the check for "fully inferred" was added.